docs/guide/snaps-concepts.md

```diff
@@ -0,0 +1,125 @@
+
+### How to derive keys
+
+To derive a user's private keys, you need to:
+
+1. Figure out whether you'll be using the BIP-32 or BIP-44 specifications to derive the user's private keys. BIP-44 is more strict in the structure and shape of the derivation paths, while BIP-32 allows for more flexibility.
+2. Find out the derivation path that you need to use. For example, if you're trying to derive keys for Dogecoin, the path is of the form `m/44'/3'/0'/0/{account_index}`.
+3. Add the required permission to your manifest file
+4. Write code in your snap to derive the keys. This will usually be done using the `@metamask/key-tree` NPM package. Any further code to e.g. derive addresses from the keys will be custom code that depends on your application.
+
+### Figuring out whether to use BIP-32 or BIP-44
+
+If the keys you are trying to derive conform to the BIP-44 structure, that is:
+
+```
+m / purpose' / coin_type' / account' / change / address_index
+```
+
+Then you should use [`snap_getBip44Entropy`](./snaps-rpc-api.html#snap-getbip44entropy) to derive your keys. Its permissions are simpler, since it requires only a coin type.
+
+If the key ayou are trying to derive do not conform to the BIP-44 structure, then you should use [`snap_getBip32Entropy`].
+
+### Finding out the derivation path
+
+The derivation path is completely dependent on the application you're building. As an example, if you're trying to reproduce the Dogecoin derivation path, that would be:
+
+```
+m/44'/3'/0'/0/{account_index}
+```
+
+This means that you'd be using [`snap_getBip44Entropy`](./snaps-rpc-api.html#snap-getbip44entropy) with the permission `coinType` having a value of `60`.
+
+### Adding permissions to the manifest file
+
+For `snap_getBip44Entropy`, you only need to specify the `coinType`, like this:
+
+```json
+{
+  "initialPermissions": {
+    "snap_getBip44Entropy": [
+      {
+        "coinType": 3 // 3 is Dogecoin
+      }
+    ]
+  }
+}
+```
+
+The authoritative list of coin types is defined in [`SLIP-44`](https://github.com/satoshilabs/slips/blob/master/slip-0044.md)
+
+### Example of private key derivation
+
+As an example, if you're trying to derive Dogecoin keys:
+
+1. Dogecoin uses the BIP-44 scheme so we'll be using [`snap_getBip44Entropy`]
+2. Dogecoin has coin type `3`, so our manifest file will have the following:
+
+   ```json
+   {
+     "initialPermissions": {
+       "snap_getBip44Entropy": [
+         {
+           "coinType": 3
+         }
+       ]
+     }
+     // other values
+   }
+   ```
+
+3. Dogecoin uses the following derivation path:
+
+   ```
+   m/44'/3'/0'/0/{account_index}
+   ```
+
+   To get the second Dogecoin account, we would write the following code:
+
+   ```javascript
+   import { getBIP44AddressKeyDeriver } from '@metamask/key-tree';
+
+   // Get the Dogecoin node, corresponding to the path m/44'/3'
+   const dogecoinNode = await wallet.request({
```

Comment:
I don't think `snap.request` is released yet, but we should either use that here and wait with merging this PR, or make a note to update this.

```suggestion
   const dogecoinNode = await snap.request({
```